packages/solecs/src/OwnableWritable.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+import { IOwnableWritable } from "./interfaces/IOwnableWritable.sol";
+
+import { Ownable } from "./Ownable.sol";
+import { OwnableWritableStorage } from "./OwnableWritableStorage.sol";
+
+/**
+ * Ownable with authorized writers
+ */
+abstract contract OwnableWritable is IOwnableWritable, Ownable {
+  error OwnableWritable__NotWriter();
+
+  /** Whether given operator has write access */
+  function writeAccess(address operator) public view returns (bool) {
+    return OwnableWritableStorage.layout().writeAccess[operator] || operator == owner();
+  }
+
+  /** Revert if caller does not have write access to this component */
+  modifier onlyWriter() {
+    if (!writeAccess(msg.sender)) {
+      revert OwnableWritable__NotWriter();
+    }
```

Comment:
curious about your preference of `revert` over `require` - could you elaborate?

Reply:
it's more about my preference of [custom errors](https://docs.soliditylang.org/en/latest/abi-spec.html#errors) over error strings. Custom errors just don't support require, unfortunately.
I took the specific semantics (Contract__ErrorName) from solidstate.

- Unlike strings, custom errors are guaranteed to be devoid of typos, since you can define them in one place, and use (via imports or inheritance) everywhere
- They allow args (like events), which is the most convenient way to do dynamic errors imo
- They aren't strings, so they should be more gas-efficient
- I just don't like the hardcoded nature of error-strings (this is similar to the 1st point, but about personal preference rather than some objective benefit)

imo it'd be great if mud started leaning towards these event-like errors. But if you ultimately prefer reverts, I can keep that in mind for future PRs

Reply:
Seconding request for use of Custom Errors moving forward!

Reply:
Thanks for elaborating, I agree custom errors are a better approach!